src/vm/loaderallocator.cpp

```diff
@@ -1236,6 +1237,8 @@ void LoaderAllocator::Init(BaseDomain *pDomain, BYTE *pExecutableHeapMemory)
         m_interopDataHash.Init(0, NULL, false, &lock);
     }
 #endif // FEATURE_COMINTEROP
+
+    m_UnmanagedCache.InitializeTable(this, &m_LoaderAllocatorUnmanagedCacheCrst);
```

Comment:
Can we move this up closer to where the `m_LoaderAllocatorUnmanagedCacheCrst` is being initialized or does this need to be done last/after this class is fully constructed? If the latter then I suggest a comment indicating this table initialization must be done at the end so it doesn't get moved.

Reply:
I wanted to do that after the heaps are created. And it also matches the way this function is structured - all the critical sections are created at the beginning together, then the heaps are created that some of the caches use and finally the caches are initialized.